src/tests/JIT/Performance/JittedMethodsCountingTest/JittedMethodsCountingTest.cs

```diff
@@ -0,0 +1,108 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+public class JittedMethodsCountingTest
+{
+    private const int MAX_JITTED_METHODS_ACCEPTED = 50;
+
+    [Fact]
+    public static int TestEntryPoint()
+    {
+        string appName = "HelloWorld.dll";
+        string jitOutputFile = "jits.txt";
+
+        // For adding any new apps for this test, make sure their success return
+        // code is 0, so we can universally handle when they fail.
+        int appResult = RunHelloWorldApp(appName, jitOutputFile);
+
+        if (appResult != 0)
+        {
+            Console.WriteLine("App failed somewhere and so we can't proceed with"
+                              + " the Jitted Methods analysis. Exiting...");
+            return 101;
+        }
+
+        // App finished successfully. We can now take a look at the methods that
+        // got jitted at runtime.
+        int jits = GetNumberOfJittedMethods(jitOutputFile);
+        return jits > 0 && jits <= MAX_JITTED_METHODS_ACCEPTED ? 100 : 101;
+    }
+
+    private static int RunHelloWorldApp(string appName, string jitOutputFile)
+    {
+        int appExitCode = -1;
+
+        // Set up our CoreRun call.
+        string coreRoot = Environment.GetEnvironmentVariable("CORE_ROOT");
+        string exeSuffix = OperatingSystem.IsWindows() ? ".exe" : "";
+        string coreRun = Path.Combine(coreRoot, $"corerun{exeSuffix}");
+
+        using (Process app = new Process())
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = coreRun,
+                Arguments = appName,
+                CreateNoWindow = true,
+                UseShellExecute = false,
+            };
+
+            // Set up the environment for running the test app. We are looking
+            // to seeing how many methods were jitted at runtime. So, we ask
+            // the runtime to print them out with DOTNET_JitDisasmSummary, and
+            // write them out to a file we can parse and investigate later
+            // with DOTNET_JitStdOutFile.
+
+            startInfo.EnvironmentVariables.Add("DOTNET_JitDisasmSummary", 1);
+            startInfo.EnvironmentVariables.Add("DOTNET_TieredCompilation", 0);
```

Comment:
Why are we disabling tiered compilation for this test?

Reply:
The HelloWorld test should be very short, perhaps we could monitor both. I agree it doesn't make much sense to run it under stress.

Reply:
The runtime tests are run in multiple configurations (tiered compilation on/off, etc.). The test itself should not need to explicitly cover different configurations.

Reply:
Sounds reasonable, in such case it should suffice to remove this environment variable override and the child process (```corerun```) should automatically inherit the parent settings.

Reply:
Sure, you should be able to achieve that by putting the property ```<CLRTestTargetUnsupported Condition="'$(TestBuildMode)' != 'crossgen2'">true</CLRTestTargetUnsupported>``` in the ```JittedMethodsCountingTest.csproj``` script. Please note that in such case you also need to mark the test as ```<RequiresProcessIsolation>true</RequiresProcessIsolation>``` as today merged wrapper generator isn't smart enough to generate a variant wrapper capable of identifying tests valid for a particular execution mode. As the number of such tests is generally low, for now we haven't tried to devise a special mechanism for them.

Reply:
(Alternatively, you can just keep the setting ```DOTNET_ReadyToRun=1``` and run the test in both R2R and JITted pipeline configurations, it's probably not going to be a fundamental contributor to our total test running time. In such case you'd need to mark the ```HelloWorld``` test with the property ```<AlwaysUseCrossGen2>true</AlwaysUseCrossGen2>``` so that it always crossgens the test no matter what the ambient setting is.

Reply:
I was thinking on this yesterday at home, and I came up with another alternative I think might be cleaner. How about checking for `DOTNET_ReadyToRun` at the beginning of the test, and if it's not enabled, end it right there returning _100_? Also, print out a message saying the test is not going to run because it's not supported outside R2R-enabled contexts. What are your thoughts on this Tomas?

Reply:
That sounds good to me, in fact it's likely going to be faster in the lab because this way your test can run in process of the merged wrapper and it will just finish immediately in the absence of ```DOTNET_ReadyToRun```. Please just remember that the default is 1 so that you need to test whether it's been explicitly overridden to 0.

Reply:
Sounds good. I was thinking maybe check whether it's different from 1 instead?

Reply:
Whatever you prefer, just keep in mind that "" (the variable not being set) means "1", not "0".

Reply:
Alright got it. I'll then make the test run only if `DOTNET_ReadyToRun` is set to "1" or not set at all, since that implicitly means "1" as well.